lib/src/discovery.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:io";
+
+import "package:path/path.dart" as path;
```

Comment:
Single letter names are also discouraged. I'd prefer to keep it unless I have an actual conflict.

Reply:
There was an actual conflict an another file. I actually feel quite strongly about this one because `path` (the package) and `String` have some shared apis. It is really easy to introduce bugs by using this name.

Reply:
OK, changed to `p`.